feature_engine/encoding/decision_tree.py

```diff
@@ -107,6 +107,18 @@ class DecisionTreeEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
         DecisionTreeClassifier(). For reproducibility it is recommended to set
         the random_state to an integer.
 
+    unseen: str, default='raise'
+        The unseen param of the OrdinalEncoder used before DecisionTreeDiscretiser
+        in the fit method. It tells the encoder how to handle unseen categories.
+        Following OrdinalEncoder we have :
+
+            -  If 'raise', then unseen categories will raise an error.
+
+            -  If 'ignore', then unseen categories will be encoded as NaN
+               and a warning will be raised instead.
+
+            -  If 'encode', unseen categories will be encoded as -1.
```

Comment:
Is this true? it will be replaced by -1 by the ordinal encoder, but the tree encoder creates a tree based on the ordinal numbers, so I am not sure this is what will happen. Did you test that?

We'd need a test for this parameter in the test, that it is able to take the strings and error when passing something else. And, ideally we also want another test to corroborate that the unseen functionality works as expected.

Reply:
Answering first to your second point point :
* Yes I will create some tests for that! 

Answering to your first point: 
* I tested that and the tree encoder encodes the unseen ones in the same "encoding value"

EX:
```python 

df_enc = pd.DataFrame ( {
        "var_A": ["A"] * 6 + ["B"] * 10 + ["C"] * 4,
        "var_B": ["A"] * 10 + ["B"] * 6 + ["C"] * 4,
        "target": [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0],
    })

variables_ = ["var_A", "var_B"]
param_grid = {"max_depth": [1, 2, 3, 4]}
X = df_enc[variables_]
y = df_enc['target']
unseen = 'raise'

# initialize categorical encoder
cat_encoder_encode = OrdinalEncoder(
    encoding_method="arbitrary",#self.encoding_method,
    variables=variables_,
    missing_values="raise",
    ignore_format=False, #self.ignore_format,
    unseen="encode" #self.unseen,
)

# initialize decision tree discretiser
tree_discretiser = DecisionTreeDiscretiser(
    cv=2, #self.cv,
    scoring="neg_mean_squared_error", #self.scoring,
    variables=variables_,
    param_grid=param_grid,
    regression=False, #self.regression,
    random_state=None #self.random_state,
)

# pipeline for the encoder
encoder_encode = Pipeline(
    [
        ("categorical_encoder", cat_encoder_encode),
        ("tree_discretiser", tree_discretiser),
    ]
)

encoder_encode.fit(X,y)

X_new = pd.DataFrame({
    "var_A":['A','B','C','ZZZ','YYY',-1],
    "var_B":['A','B','C','A'  ,'ZZZ',-1],
})

encoder_encode.transform(X_new)

>>>>
>>>>
>>>>
>>>>

```

Reply:
Answering first to your second point point :
* Sure I will create a test for the possible values of the param
* And a test to check that work as expected  

Answering to your first point: 
* I quickly tested, the tree encoder encodes all the unseen values in the same "encoding bucket". I am providing here a code snippet with the output. For the `var_A` it seems 0.25 the 'Other' bucket on which encode some categories and unseen values.

EX:
```python 

df_enc = pd.DataFrame ( {
        "var_A": ["A"] * 6 + ["B"] * 10 + ["C"] * 4,
        "var_B": ["A"] * 10 + ["B"] * 6 + ["C"] * 4,
        "target": [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0],
    })

variables_ = ["var_A", "var_B"]
param_grid = {"max_depth": [1, 2, 3, 4]}
X = df_enc[variables_]
y = df_enc['target']
unseen = 'raise'

# initialize categorical encoder
cat_encoder_encode = OrdinalEncoder(
    encoding_method="arbitrary",#self.encoding_method,
    variables=variables_,
    missing_values="raise",
    ignore_format=False, #self.ignore_format,
    unseen="encode" #self.unseen,
)

# initialize decision tree discretiser
tree_discretiser = DecisionTreeDiscretiser(
    cv=2, #self.cv,
    scoring="neg_mean_squared_error", #self.scoring,
    variables=variables_,
    param_grid=param_grid,
    regression=False, #self.regression,
    random_state=None #self.random_state,
)

# pipeline for the encoder
encoder_encode = Pipeline(
    [
        ("categorical_encoder", cat_encoder_encode),
        ("tree_discretiser", tree_discretiser),
    ]
)

encoder_encode.fit(X,y)

X_new = pd.DataFrame({
    "var_A":['A','B','C','ZZZ','YYY',-1],
    "var_B":['A','B','C','A'  ,'ZZZ',-1],
})

encoder_encode.transform(X_new)

>>>> var_A var_B
>>>> 0.25  0.2
>>>> 0.25  0.2
>>>> 0.50  0.4
>>>> 0.25  0.2
>>>> 0.25  0.2
>>>> 0.25  0.2 

```

Reply:
Doing further checks raise both "raise" and "ignore" raise an exception.
"raise" --> it raises a ValueError for new values
"ignore" --> It passes the first part of the pipeline `OrdinalEncoder` (generating Nans for unseen) but it throws an error in the second part for the `DecisionTreeDiscretiser` because it is fed with Nan values which `DecisionTreeDiscretiser` does not allow .

I am planning to change the  `DecisionTreeDiscretiser` class to allow only `raise` and `encode` for the `unseen` param